tests/test_widget.py

```diff
@@ -133,3 +186,53 @@ def test_calculate_length_fails_with_wrong_number_of_points(
         "ERROR: Select two points to calculate the decay length."
         in captured.out
     )
+
+
+@pytest.mark.parametrize(
+    "data_subdirs, image_count, data_loaded",
+    [
+        (["my_view1", "my_view2", "my_view3"], [2, 2, 2], True),
+        (["my_view1", "my_view2"], [2, 2], False),
+        (["my_view1", "my_view2", "my_view3"], [1, 2, 2], False),
+        (["my_view1", "my_view2", "no_view"], [2, 2, 2], False),
+    ],
+)
+def test_load_data(
+    cpt_widget, capsys, tmp_path, qtbot, data_subdirs, image_count, data_loaded
+):
+    """Test loading of images in a folder as stack associated to a certain view"""
+
+    for subdir, n in zip(data_subdirs, image_count):
+        p = tmp_path / subdir
+        p.mkdir()
+        for i in range(n):
+            data = np.random.randint(0, 255, (256, 256), "uint8")
+            tf.imwrite(p / f"temp{i}.tif", data)
+
+    def set_directory_and_close(dialog):
+        qtbot.addWidget(dialog)
+        dialog.setDirectory(str(tmp_path))
+        buttonbox = dialog.findChild(QDialogButtonBox, "buttonBox")
+        openbutton = buttonbox.children()[1]
+        qtbot.mouseClick(openbutton, Qt.LeftButton, delay=1)
+
+    # Open and retrieve file dialog
+    get_dialog(
+        dialog_trigger=cpt_widget._on_click_load_data,
+        dialog_action=set_directory_and_close,
+        time_out=5,
+    )
+
+    if data_loaded:
+        assert len(cpt_widget.viewer.layers) == 3
+        for i in range(3):
+            assert (
+                cpt_widget.viewer.layers[i].name == "stack" + str(i + 1)
+                and cpt_widget.viewer.layers[i].ndim == 3
+            )
+    else:
+        captured = capsys.readouterr()
+        assert (
+            "WARNING: The data folder must contain three subfolders, one for each view, and each subfolder must contain the same number of images."
+            in captured.out
```

Comment:
I guess we want to have the napari error message eventually. Future ticket to unify the error output to the user?

- #27